core/src/main/java/edu/wpi/grip/core/SwitchableSource.java

```diff
@@ -0,0 +1,23 @@
+package edu.wpi.grip.core;
+
+
+import java.io.IOException;
+
+/**
+ * A {@link Source} that can switch its value.
+ */
+public abstract class SwitchableSource extends Source {
+
+    /**
+     * Loads the next value that this source has into the socket.
+     * @throws IOException If the source fails to load the next value into the socket.
+     */
+    public abstract void nextValue() throws IOException;
```

Comment:
How would you select a particular source with this?  Like, if I want to go to the tenth one.

Overall, I don't think there's any use right now in making this its own abstract class.  Just put the methods in the image source, so we can change its functionality without also changing this or adding a new base class.